### ISSUE TITLE
"E117: Unknown function: uniq" in gutentags#setup_gutentags..gutentags#get_project_root

### ISSUE THREAD

**Issue by smkent** [ludovicchabant/vim-gutentags#89]
https://github.com/ludovicchabant/vim-gutentags/commit/270f645d9df72cc21a36d7f3e5675bb155571434 adds a call to `uniq` which breaks versions of vim older than 7.4.218:

```
Error detected while processing function gutentags#setup_gutentags..gutentags#get_project_root:
line   11:
E117: Unknown function: uniq
```

Ubuntu 14.04 (LTS) ships vim 7.4.52, which doesn't have `uniq`. vim-gutentags should either not use `uniq` or use a backwards compatible approach ([examples in this issue](https://github.com/LaTeX-Box-Team/LaTeX-Box/issues/190)).

**Comment by smkent** [comment]
If you want to use `uniq` on newer versions of vim while skipping the duplicate check on older versions, you could use a simple workaround such as:

```vim
if exists('*uniq')
    let l:markers = uniq(sort(l:markers))
endif
```

**Comment by ludovicchabant** [comment]
Ah damn, thanks for reporting that! Should be fixed with cf9a4ebd1bc102d033787e7b1b3b25cae81c893c.

I'm not sure how you can tell that `uniq` only exists in recent versions? `:help uniq()` doesn't say much -- there's just a cryptic `E882` in the entry header which I don't really know what it means?

**Comment by smkent** [comment]
Thanks for the quick fix! I updated to latest and my vim is happy again.

The machine I'm currently on is running Ubuntu 14.04 with vim 7.4.52, and the `uniq` function doesn't exist in this version of vim which is how I discovered the issue. I found 7.4.218 as the version where `uniq` was added by searching for "unknown function uniq," it seems a few other vim plugins have run into the same issue. [I also found on the vim-dev mailing list where `uniq` was introduced in patch 218.](https://groups.google.com/forum/#!topic/vim_dev/iwws3jIa39s)
